File: netease-nimsdk/src/main/java/com/netease/nim/uikit/business/session/actions/TakePictureAction.java
package com.netease.nim.uikit.business.session.actions;

import com.netease.nim.uikit.common.media.imagepicker.ImagePickerLauncher;
import com.netease.nimlib.sdk.chatroom.ChatRoomMessageBuilder;
import com.netease.nimlib.sdk.msg.MessageBuilder;
import com.netease.nimlib.sdk.msg.constant.SessionTypeEnum;
import com.netease.nimlib.sdk.msg.model.IMMessage;
import com.zxn.netease.nimsdk.R;

import java.io.File;

/**
 * Created by zxn on 2020/10/20.
 */
public class TakePictureAction extends PickImageAction{

    public TakePictureAction() {
        super(R.drawable.nim_message_plus_video_selector, R.string.input_panel_take, true);
    }

    @Override
    protected void showSelector(int titleId, int requestCode, boolean multiSelect) {
        //super.showSelector(titleId, requestCode, multiSelect);
        ImagePickerLauncher.takePhoto(getActivity(),requestCode);
    }

    @Override
    protected void onPicked(File file) {
        IMMessage message;
        if (getContainer() != null && getContainer().sessionType == SessionTypeEnum.ChatRoom) {
            message = ChatRoomMessageBuilder.createChatRoomImageMessage(getAccount(), file, file.getName());
        } else {
            message = MessageBuilder.createImageMessage(getAccount(), getSessionType(), file, file.getName());
        }
        sendMessage(message);
    }
}

File: demo/src/main/java/com/netease/nim/demo/session/action/TakePictureAction.kt
package com.netease.nim.demo.session.action

import com.netease.nimlib.sdk.msg.MessageBuilder
import com.zxn.netease.nimsdk.R
import com.zxn.netease.nimsdk.business.session.actions.PickImageAction
import com.zxn.netease.nimsdk.common.media.imagepicker.ImagePickerLauncher
import java.io.File

/**
 * Created by zxn on 2020/10/20.
 */
open class TakePictureAction :
    PickImageAction(R.drawable.nim_message_plus_video_selector, R.string.input_panel_take, true) {
    override fun showSelector(titleId: Int, requestCode: Int, multiSelect: Boolean) {
        ImagePickerLauncher.takePhoto(activity, requestCode)
    }

    override fun onPicked(file: File?) {
        file?.let {
            val message = MessageBuilder.createImageMessage(account, sessionType, file, file.name)
            sendMessage(message)
        }
    }
}